comp/metadata/packagesigning/packagesigningimpl/aptsigning.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package packagesigningimpl
+
+import (
+	"bufio"
+	"encoding/xml"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	pkgUtils "github.com/DataDog/datadog-agent/comp/metadata/packagesigning/utils"
+)
+
+const (
+	trustedFolder  = "/etc/apt/trusted.gpg.d/"
+	trustedFile    = "/etc/apt/trusted.gpg"
+	mainSourceList = "/etc/apt/sources.list"
+	sourceList     = "/etc/apt/sources.list.d/"
+)
+
+var (
+	sourceListRegexp = regexp.MustCompile(`^([^\s]+)\s?(\[.*\]\s)?(.*)$`)
+	signedBy         = regexp.MustCompile(`signed-by=([A-Za-z0-9_\-\.\/]+)`)
+	debsigPolicies   = "/etc/debsig/policies/"
+	debsigKeyring    = "/usr/share/debsig/keyrings/"
+)
+
+// getAPTSignatureKeys returns the list of debian signature keys
+func getAPTSignatureKeys(client *http.Client, logger log.Component) []SigningKey {
+	allKeys := make(map[string]SigningKey)
```

Comment:
@chouetz, correct me if I'm wrong, but it looks like `alleys` is used as a cache mechanism, using the fingerprint as the key inside the `decryptGPGReader` function. I assume that we use a map for the ease of checking if a value is present or not, but ultimately we end up discarding the key values when iterating over `allKeys` and only return the `[]SigningKey`

Maybe we could benefit from renaming `allKeys` to something else, something that reflects that it is used for caching `cacheKeys`  or `readKeys`

Reply:
yes it's to prevent duplication (and I'm changing the index right now to use fingerprint + keyType), I can rename as cacheKeys 👍🏽

Reply:
Thanks for the confirming 😄 and for accepting the suggestion